libraries/AP_Mount/AP_Mount_Alexmos.cpp

```diff
@@ -26,72 +26,85 @@ void AP_Mount_Alexmos::update()
 
     read_incoming(); // read the incoming messages from the gimbal
 
+    // contains true if target is valid
+    bool isTargetValid = false;
+
     // update based on mount mode
     switch (get_mode()) {
         // move mount to a "retracted" position.  we do not implement a separate servo based retract mechanism
         case MAV_MOUNT_MODE_RETRACT: {
             const Vector3f &target = _params.retract_angles.get();
-            _angle_rad.roll = radians(target.x);
-            _angle_rad.pitch = radians(target.y);
-            _angle_rad.yaw = radians(target.z);
-            _angle_rad.yaw_is_ef = false;
+            mnt_target.target_type = MountTargetType::ANGLE;
+            mnt_target.angle_rad.set(ToRad(target.x), ToRad(target.y), ToRad(target.z), false);
+            isTargetValid = true;
             break;
         }
 
         // move mount to a neutral position, typically pointing forward
         case MAV_MOUNT_MODE_NEUTRAL: {
             const Vector3f &target = _params.neutral_angles.get();
-            _angle_rad.roll = radians(target.x);
-            _angle_rad.pitch = radians(target.y);
-            _angle_rad.yaw = radians(target.z);
-            _angle_rad.yaw_is_ef = false;
+            mnt_target.target_type = MountTargetType::ANGLE;
+            mnt_target.angle_rad.set(ToRad(target.x), ToRad(target.y), ToRad(target.z), false);
+            isTargetValid = true;
             break;
         }
 
         // point to the angles given by a mavlink message
         case MAV_MOUNT_MODE_MAVLINK_TARGETING:
-            switch (mavt_target.target_type) {
-            case MountTargetType::ANGLE:
-                _angle_rad = mavt_target.angle_rad;
-                break;
-            case MountTargetType::RATE:
-                update_angle_target_from_rate(mavt_target.rate_rads, _angle_rad);
-                break;
-            }
+            // mavlink targets are stored while handling the incoming message and are considered valid
+            isTargetValid = true;
             break;
 
         // RC radio manual angle control, but with stabilization from the AHRS
         case MAV_MOUNT_MODE_RC_TARGETING: {
             // update targets using pilot's RC inputs
-            MountTarget rc_target {};
-            if (get_rc_rate_target(rc_target)) {
-                update_angle_target_from_rate(rc_target, _angle_rad);
-            } else if (get_rc_angle_target(rc_target)) {
-                _angle_rad = rc_target;
+            if (get_rc_rate_target(mnt_target.rate_rads)) {
+                isTargetValid = true;
+                mnt_target.target_type = MountTargetType::RATE;
```

Comment:
Any reason not to have a `MountTargetType::INVALID` ?

You can then switch on it rather than checking for validity and then comparing